docs/source/the_quantum_computer.rst

```diff
@@ -407,8 +250,62 @@ the named QPU. It's a good idea to run your programs against the QVM before book
 bugs. To learn more about how to add noise models to your virtual ``QuantumComputer`` instance, check out
 :ref:`noise`.
 
-In the next section, we will see how to use the Wavefunction Simulator aspect of the QVM server to inspect the full
-wavefunction set up by a Quil program.
+Differences between a QVM and a QPU based ``QuantumComputer``
+=============================================================
 
-.. [1] https://arxiv.org/abs/1608.03355
+As mentioned above, pyQuil is designed such that code based on a ``QuantumComputer`` can be used in more or less the same way,
+regardless if it is based on a QVM or QPU. However, depending on which you are using, the subcompoments have additional features
+worth knowing about.
+
+For instance, if your code targets a QVM, ``qc.qam`` will be a :py:class:`~pyquil.api.QVM``` instance, and ``qc.compiler`` will
+be a :py:class:`~pyquil.api.QVMCompiler` instance. However, if your code targets a QPU, ``qc.qam`` will be a :py:class:`~pyquil.api.QPU` instance, and ``qc.compiler`` will be a :py:class:`~pyquil.api.QPUCompiler` instance.
+
+While these subcomponents follow common interfaces, namely :py:class:`~pyquil.api.QAM` and
+:py:class:`~pyquil.api.AbstractCompiler`, there may be some methods or properties that are accessible on the QPU-based instances
+but not on the QVM-based instances, and vice versa.
+
+You can access these features and keep your code robust by performing type checks on ``qc.qam`` and/or ``qc.compiler``.
```

Comment:
Good tip.

Reply:
Lifted from the GitBook, possibly a tip you wrote long ago? 😂